content/article/the-rock-pi-s-or-how-to-get-gpio-input-without-a-library.md

```diff
@@ -0,0 +1,198 @@
+
+  {
+    "title"       : "The Rock Pi S; or, How to Get GPIO Input Without a Library",
+    "authors"     : ["timm-murray"],
+    "date"        : "2020-01-20T17:50:11",
+    "tags"        : [ "hardware", "linux", "gpio", "pi", "rockpi" ],
+    "draft"       : true,
+    "image"       : "/images/the-rock-pi-s-or-how-to-get-gpio-input-without-a-library/rockpi.jpg",
+    "thumbnail"   : "",
+    "description" : "There are lots of single board computers out there, but not all of them have an simple library for getting input on a pin. Never fret, because most of them run Linux, and it's easy to access pins with just Linux.",
+    "categories"  : "hardware"
+  }
+
+When writing articles like these, there is an inevitable comparison to the 
+Raspberry Pi series. There is no way to fight this, 
+and for good reason. The first Raspberry Pi ushered in a slew of Single 
+Board Computers (SBCs), and one of them is the Rock Pi S, a new board from Seeed 
+Studio.
+
+In the interest of full disclosure, the Rock Pi S provided by Seeed 
+Studio for this review. Specifically, it's the model with 512MB of RAM and 
+4Gb of built-in flash.
+
+The Rock Pi S competes in the same segment as the Raspberry Pi Zero, particularly the Zero W 
+with built-in WiFi. Its form factor is different; where the Zero is 
+shaped like a stick of gum, the Rock Pi S is closer to a square. The Rock 
+also has USB-C for power, an ethernet jack, and a USB-A port. Depending on the 
+project, it can end up being cheaper than the Zero, since you don't have to 
+buy a micro-USB to USB-A adapter to hook up most other devices.
+
+You do still need a micro SD card. While there are versions of the Rock with 
+built-in flash, it's small and not meant for booting an OS. Note that the size of the built-in flash
+is listed in giga*bits*. The 4Gb version is actually 512 mega*bytes*. So get an 
+SD card, preferably one with an A1 rating, which specifies that it works well 
```

Comment:
Ah, did not know this! Is there a resource you can link to for the right sort of card? Which particular one did you use?